Tests/CalculatorApp.Tests/InputParsingTests.cs

```diff
@@ -0,0 +1,147 @@
+using Xunit;
+
+namespace CalculatorApp.Tests;
+
+public class InputParsingTests
+{
+    private readonly Calculator _calculator;
+
+    public InputParsingTests()
+    {
+        _calculator = new Calculator();
+    }
+
+    [Theory]
+    [InlineData("  2 + 3  ", 5)]
+    [InlineData("2+   3", 5)]
+    [InlineData("  (2+3)*4  ", 20)]
```

Comment:
add a case where a negative number has whitespace between the negative sign and the number

```suggestion
    [InlineData("  (2+3)*4  ", 20)]
    [InlineData("-   5+0", -5)]
```